### ISSUE TITLE
Faraday::Request::Retry default exceptions as a constant

### ISSUE THREAD

**Issue by krmannix** [lostisland/faraday#813]
Not reporting a bug or issue, so eschewing the template per instructions.

We want to add a few extra exceptions to the default list of exceptions provided by `Request::Retry`. However, this leads us to using `Faraday::Request::Retry::Options.new.exceptions` to fetch the default list of exceptions to work with. Alternatively, creating a custom list of retriable methods in addition to the default list seems quite natural (`Faraday::Request::Retry::IDEMPOTENT_METHODS`).

I'd like to make the list of default exceptions a constant on `Faraday::Request::Retry`, but per contribution instructions wanted to create an issue here first in case this has already been discussed.

My implementation would result in custom exceptions arguments like below:

```
Faraday::Connection.new(url, options) do |conn|
  conn.request :retry, max: 1, interval: 0.5,
                       exceptions: Faraday::Request::Retry::EXCEPTIONS + [Faraday::ConnectionFailed]
end
```

**Comment by iMacTia** [comment]
Hi @krmannix, makes sense to me, thanks for the contribution!

The only thing I would change is the name of the constant into `DEFAULT_EXCEPTIONS` to make it very clear it's just the default value, as we do for `DEFAULT_CHECK`.

**Comment by iMacTia** [comment]
Closed by #814
